platform/viewer/cypress/support/commands.js

```diff
@@ -41,17 +41,45 @@ import {
 Cypress.Commands.add('openStudy', patientName => {
   cy.initRouteAliases();
   cy.visit('/');
-  cy.get('#patientName').type(patientName);
+  cy.wrap(null).then(() => {
+    return new Cypress.Promise((resolve, reject) => {
+      cy.get('#patientName').type(patientName);
+      setTimeout(() => {
+        cy.get('#studyListData')
+          .contains(patientName)
+          .first()
+          .click();
+        resolve();
+      }, 2000);
+    });
+  });
+});
 
-  cy.get('.studylistStudy > .patientName', { timeout: 5000 })
-    .contains(patientName)
+/**
+ * Command to search for a modality and open the study.
+ *
+ * @param {string} modality - Modality type that we would like to search for
+ */
+Cypress.Commands.add('openStudyModality', modality => {
+  cy.initRouteAliases();
+  cy.visit('/');
+  cy.get('#modalities')
+    .type(modality)
+    .wait(2000);
```

Comment:
Instead of an explicit wait here, we probably want a timeout for the below `cy.get` to contain the modality. If that's failing, it could be related to the on blur issue we identified earlier, but that it a "bug" (as far as I'm concerned) that we should work to resolve.

In general, if we have an explicit wait, we should try to identify why, as it's usually an indicator that we have something we need to create an issue for and resolve on the development side.

Reply:
We can say that the problem here to the asynchronism. If we only add the `timeout` on the `cy.get('#studyListData')`, we will either get an empty element (since the element is present in the DOM) or a full element (a full list of the patients listed in that page). If we get an empty element, it will definitely fail. If we get the full page list, it may fail (if the searched patient were not initially listed on the 1st page) or not. This may occur failures if we change/add more patient data.

With this `2s` wait, we are making sure that the captured element will contain the search result.